<a href="https://colab.research.google.com/github/MCn21thCntry/Practical-Machine-Learning---from-the-rooter-to-the-tooter/blob/main/Intro_ML_Assignment_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Module 2 Homework: Exploring Housing Data in Iowa
---

**`Objective`**: This homework is designed to reinforce your understanding of basic data exploration techniques using Pandas, as covered in Module 2. You will be working with the Iowa Housing dataset to apply the methods you've learned to understand its structure and key characteristics.

**`Dataset`**: We will be using the Iowa Housing dataset, which you used in Module 2. If you are having trouble accessing it through Kaggle in Colab, you can also find a version of it on platforms like GitHub by searching for "Iowa Housing Dataset CSV".

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
# Import necessary libraries: Import pandas as pd and numpy as np.
import pandas as pd
import numpy as np

In [3]:
# Load the Iowa Housing dataset: Use pd.read_csv() to load the train.csv file into a Pandas DataFrame named housing_data.
housing_data = pd.read_csv('/content/Iowa Housing Dataset/train.csv')



#``` Question 1: Initial Data Inspection```



In [4]:
# (Code): Display the first 7 rows of the housing_data DataFrame using the appropriate Pandas method.
housing_data.head(7)

Order        PID  MS SubClass MS Zoning  Lot Frontage  Lot Area Street  \
0    534  531363010           20        RL          80.0      9605   Pave   
1    803  906203120           20        RL          90.0     14684   Pave   
2    956  916176030           20        RL           NaN     14375   Pave   
3    460  528180130          120        RL          48.0      6472   Pave   
4    487  528290030           80        RL          61.0      9734   Pave   
5   1965  535453150           20        RL          70.0      7315   Pave   
6    681  535452090           20        RL          71.0      7350   Pave   

  Alley Lot Shape Land Contour  ... Pool Area Pool QC Fence Misc Feature  \
0   NaN       Reg          Lvl  ...         0     NaN   NaN          NaN   
1   NaN       IR1          Lvl  ...         0     NaN   NaN          NaN   
2   NaN       IR1          Lvl  ...         0     NaN   NaN          NaN   
3   NaN       Reg          Lvl  ...         0     NaN   NaN          NaN   
4   NaN       IR1          Lvl  ...         0     NaN   NaN          NaN   
5   NaN       Reg          Lvl  ...         0     NaN   NaN          NaN   
6   NaN       Reg          Lvl  ...         0     NaN  GdWo          NaN   

  Misc Val Mo Sold Yr Sold Sale Type  Sale Condition  SalePrice  
0        0       4    2009       WD           Normal     159000  
1        0       6    2009       WD           Normal     271900  
2        0       1    2009       COD         Abnorml     137500  
3        0       4    2009       WD           Normal     248500  
4        0       5    2009       WD           Normal     167000  
5        0       3    2007       WD           Normal     140000  
6        0       3    2009       WD           Normal     128000  

[7 rows x 82 columns]

## Looking at the first few rows, can you identify at least three column names that seem to represent:


*   A numerical feature related to size or area.
*   A numerical feature related to price.
*   A categorical feature (representing categories or types).

In [5]:
housing_data[['Lot Area', "SalePrice","Lot Shape"]].head(3)

Lot Area  SalePrice Lot Shape
0      9605     159000       Reg
1     14684     271900       IR1
2     14375     137500       IR1

#``` Question 2: Dataset Dimensions and Data Types```

In [6]:
# Find and print the number of rows and columns in the housing_data DataFrame.
housing_data.shape

(2197, 82)

## How many data entries (houses) are in this dataset, and how many features (characteristics) are recorded for each house? What Pandas method did you use to find this information?
# 2197 entries and 82 features are in this dataset. I used shape method to find this information.

#``` Question 3: Descriptive Statistics of 'LotArea'```

In [7]:
# Calculate and print the following descriptive statistics for the 'LotArea' column in the housing_data: Mean, Median, Standard Deviation, Minimum value, Maximum value
housing_data['Lot Area'].describe()

count      2197.000000
mean      10268.070096
std        7975.740885
min        1300.000000
25%        7500.000000
50%        9488.000000
75%       11660.000000
max      215245.000000
Name: Lot Area, dtype: float64

## Based on the statistics you calculated, briefly describe the typical 'LotArea' in this dataset. Is the distribution likely to be symmetrical or skewed? Explain your reasoning by comparing the mean and median.
It seems it is little bit positively skewed because the mean higher but not too much than median value.

#``` Question 4: Value Counts of 'HouseStyle'```

In [8]:
# Find and print the value counts for the 'HouseStyle' column in the housing_data.
housing_data['House Style'].value_counts()

House Style
1Story    1123
2Story     659
1.5Fin     219
SLvl        97
SFoyer      64
2.5Unf      17
1.5Unf      14
2.5Fin       4
Name: count, dtype: int64

## Which 'HouseStyle' is the most frequent in the dataset? What does .value_counts() tell us about the distribution of different house styles in this dataset?
1Story is the most frequent in the dataset. .value_counts() tells us 1Story, 2Story are dominantly the most common styles. 2.5Unf, 1.5Unf and 2.5Fin are so less so they can be classed as a one class.


#``` Question 5: Interpreting 'SalePrice' Statistics```

In [9]:
# Calculate and print the descriptive statistics for the 'SalePrice' column using the .describe() method.
print(housing_data['SalePrice'].describe())

count      2197.000000
mean     182376.851161
std       81168.157405
min       13100.000000
25%      130000.000000
50%      163500.000000
75%      215000.000000
max      755000.000000
Name: SalePrice, dtype: float64


## Focusing on the 'SalePrice' statistics from .describe():
What is the average (mean) 'SalePrice' in this dataset?
What is the range of 'SalePrice' values (from minimum to maximum)?
What does the standard deviation tell you about the spread of 'SalePrice' values? Is it a relatively large or small standard deviation compared to the mean, and what does this imply about the variability in house prices?

*   The mean is 182376.851161.
*   The range of 'SalePrice' is 755000.000000 - 13100.000000 = 741900.
*   The standard deviation tells me the smallest value can be around 100.000 and the biggest value can be around 260.000.
It a relatively small standard deviation compared to the mean and this implies that the values have small range of scope.




In [10]:
housing_data['SalePrice'].max() - housing_data['SalePrice'].min()

741900